### PR TITLE
Added hecto discrete grid support to RoadSegment

### PR DESCRIPTION
**Pull request by agoodman**
Added hecto_key to RoadSegment model and configured an index on it.

Exposed new endpoint  `road_segments/:factor/:key`  to enable discrete
grid querying of segment data. The factor is 1000, 10000, or 100000;
representing a multiplier on lat/lng. The key is a pair of 30-bit
base64 encoded strings;lat and lng, respectively.

Enabled NewRelic monitoring and MemCachier caching via dalli gem.

Encoding strategy described in detail here: https://cityhikerapp.com/2016/09/30/map-search-optimization/

### CODE REVIEW

**Comment by agoodman** [comment]
Hey @Nick011 or @jwd2a or @danlynn, wanna review this for me? I think you'd enjoy the subject matter.

**Comment by jwd2a** [comment]
Definitely interested in the subject matter. Also much further into smart kid territory than my current level of acumen permits me to go :P

**Comment by Nick011** [comment]
I'm not really qualified to review ruby =. I haven't done ruby since I worked with you. I can review python, js, db architecture, and api's confidently.

File: app/models/road_segment.rb [hunk 1]
```diff
@@ -13,6 +13,59 @@ def haversine
...
+    end
+
+    uint_val = Integer(value.abs * factor)
+
+    int_val = int_val | uint_val
+
+    return int_val
+  end
+  
+  @@base64_characters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z", "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "+", "-"]
```
Comment (Nick011) [inline app/models/road_segment.rb:52]: This seems a bit odd. There is this thing in python: http://stackoverflow.com/questions/5891453/is-there-a-python-library-that-contains-a-list-of-all-the-ascii-characters

Or a regex?

Comment (Nick011) [inline app/models/road_segment.rb:52]: May also want to consider putting something like this in your "settings" or "config" if you might need it else where.

Comment (agoodman) [reply app/models/road_segment.rb:52]: This is the encoding map. It's a constant array. The concept here is to take 6 bits at a time in a 30-bit sequence. Each resolves to a value representing an index in this map. So, `010011 (19)` encodes as `t`.

**Pull request merged**